tornado/websocket.py

```diff
@@ -1082,10 +1093,17 @@ def write_message(
             opcode = 0x2
         else:
             opcode = 0x1
-        message = tornado.escape.utf8(message)
-        assert isinstance(message, bytes)
-        self._message_bytes_out += len(message)
+
         flags = 0
+
+        if isinstance(message, list):
+            self._message_bytes_out += sum(len(i) for i in message)
+        else:
+            if isinstance(message, str):
+                message = tornado.escape.utf8(message)
+            assert isinstance(message, (bytes, memoryview))
+            self._message_bytes_out += len(message)
```

Comment:
This length computation is not correct for memorviews AFAIK:
```
In [13]: a = np.linspace(0,1, 100)

In [15]: mv = memoryview(a)

In [16]: len(mv)
Out[16]: 100

In [17]: mv.nbytes
Out[17]: 800
```

Reply:
This bug is present since Tornado 4.5: `BaseIOStream.write()` mention that from version 4.5 it supports memoryviews. actually not :) because of their shape.

Anyway. I've fixed also this issue.

Reply:
Great, I will take a look tonight